File: wisefy/src/androidTest/java/com/isupatches/wisefy/IsDeviceConnectedToMobileOrWifiNetworkTests.java
package com.isupatches.wisefy;


import android.net.NetworkInfo;
import com.isupatches.wisefy.base.TestActivity;
import org.junit.Test;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;


public class IsDeviceConnectedToMobileOrWifiNetworkTests extends BaseAndroidJUnit4TestClass {

    @Test
    public void failure_missingPrerequisite() {
        missingPrerequisite();
        assertFalse(mWiseFy.isDeviceConnectedToMobileOrWifiNetwork());
    }

    @Test
    public void failure() {
        when(mMockConnectivityManager.getActiveNetworkInfo()).thenReturn(mock(NetworkInfo.class));

        WiseFyConnection mockWiseFyConnection = mock(WiseFyConnection.class);
        mWiseFy.mWiseFyConnection = mockWiseFyConnection;
        when(mockWiseFyConnection.isNetworkConnected(any(NetworkInfo.class))).thenReturn(false);
        assertFalse(mWiseFy.isDeviceConnectedToMobileOrWifiNetwork());
    }

    @Test
    public void success() {
        when(mMockConnectivityManager.getActiveNetworkInfo()).thenReturn(mock(NetworkInfo.class));

        WiseFyConnection mockWiseFyConnection = mock(WiseFyConnection.class);
        mWiseFy.mWiseFyConnection = mockWiseFyConnection;
        when(mockWiseFyConnection.isNetworkConnected(any(NetworkInfo.class))).thenReturn(true);
        assertTrue(mWiseFy.isDeviceConnectedToMobileOrWifiNetwork());
    }
}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/IsDeviceConnectedToMobileOrWifiNetworkTests.kt
package com.isupatches.wisefy

import com.isupatches.wisefy.internal.base.BaseInstrumentationTest

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

/**
 * Tests the ability to determine if a device is connected to a mobile or wifi network.
 *
 * @author Patches
 * @since 3.0
 */
internal class IsDeviceConnectedToMobileOrWifiNetworkTests : BaseInstrumentationTest() {

    @Test fun failure_missingPrerequisite() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToMobileOrWifiNetwork_failure()
        assertFalse(wisefy.isDeviceConnectedToMobileOrWifiNetwork())
    }

    @Test fun failure() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToMobileOrWifiNetwork_success()
        mockWiseFyConnectionUtil.isNetworkConnected(false)
        assertFalse(wisefy.isDeviceConnectedToMobileOrWifiNetwork())
    }

    @Test fun success() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToMobileOrWifiNetwork_success()
        mockNetworkUtil.activeNetwork()
        mockWiseFyConnectionUtil.isNetworkConnected(true)
        assertTrue(wisefy.isDeviceConnectedToMobileOrWifiNetwork())
    }
}
